libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -470,54 +366,61 @@ - (NSString*)uniqueUserAccountIdentifier {
     return identifier;
 }
 
+-(SFOAuthCredentials *)oauthCredentials {
+
+    NSString *oauthClientId = [self oauthClientId];
+    NSString *identifier = [self uniqueUserAccountIdentifier];
+
+    SFOAuthCredentials *creds = [[SFOAuthCredentials alloc] initWithIdentifier:identifier clientId:oauthClientId encrypted:YES];
+    creds.redirectUri = self.oauthCompletionUrl;
+    creds.domain = self.loginHost;
+    creds.accessToken = nil;
+    creds.redirectUri = self.oauthCompletionUrl;
+    creds.clientId = self.oauthClientId;
+    return creds;
+}
+
 - (SFUserAccount*)createUserAccount {
-    SFUserAccount *newAcct = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier]];
+
+    SFUserAccount *newAcct = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier] clientId:self.oauthClientId];
     SFOAuthCredentials *creds = newAcct.credentials;
     creds.accessToken = nil;
     creds.domain = self.loginHost;
     creds.redirectUri = self.oauthCompletionUrl;
     creds.clientId = self.oauthClientId;
 
-    // When creating a fresh user account, always use a default user ID
-    // and org ID until the server tells us what the actual IDs are.
-    creds.userId = self.temporaryUserIdentity.userId;
-    creds.organizationId = self.temporaryUserIdentity.orgId;
-
     //add the account to our list of possible accounts, but
     //don't set this as the current user account until somebody
     //asks us to login with this account.
-    [self addAccount:newAcct];
+    [self saveAccountForUser:newAcct error:nil];
 
     return newAcct;
 }
 
 - (SFUserAccount*)createUserAccountWithCredentials:(SFOAuthCredentials*)credentials {
+
     SFUserAccount *newAcct = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier]];
```

Comment:
With regards to my previous comments around tying the credentials identifier with the user account identifier, we're eschewing that relationship here, intentionally or unintentionally.  If that's working, it gives more credence to putting the generation of new `SFOAuthCredentials` objects under the purview of `SFAuthenticationManager`.